### ISSUE TITLE
Generate images in batch/Run program from command line

### ISSUE THREAD

**Issue by Bentroen** [Pangamma/PixelStacker#12]
Hello!

First of all, I'd like to say that the idea of using double layers is absolutely brilliant! It generates far better results than any pixel art tool I had seen before.

I'm looking for a tool that can be integrated with Python scripts to generate images in batch. In fact, there are a few Python libraries that work similarly, but they're mostly very limited and the results aren't nearly as good. So I've been thinking: could there be a way to run the program from the command line, or else, just batch generate images from inside the program itself? Would already be quite helpful. :)

Thanks in advance!

**Comment by Pangamma** [comment]
Hmmmm. Interesting proposal. What would a feature like this be used for? Gif images? Also, what would some suggested CLI arguments be?

**Comment by Bentroen** [comment]
Sorry for the delay! I had previously written an answer, but I lost it due to a power outage. :/

Yeah, that's exactly the purpose I have in mind! Perhaps even converting whole videos into Minecraft blocks. It probably wouldn't be perfect, but it's more an experiment to push the game's limits than a fully fledged concept.

As for the CLI arguments, here are some suggestions:

- Two required parameters pointing to the image to be converted, and the path where the resulting file will be saved. The export type could be detected based on the extension of the output file (`.png`, `.schematic`, `.schem` or `.csv`).
- Since there's already an option in the program to export block definitions to a JSON file, there could be a `--profile` parameter that takes the path to that file and uses it as the profile.
- A boolean flag, `--singlelayer`, that when present will deactivate the "Enable second layer" option. This preserves the default behavior, i.e. when it's not present the image will be rendered with both layers.
- A boolean flag `--sideview`, that when specified will change to side view.
- Two parameters, `--width` and `--height` specifying the max width and height. When not present, would behave as inputting 0, i.e. no max size for that dimension.
- Being able to control the quantizer from the command line would also be interesting, though I imagine it'd be a pain to do arguments for that. <_< Perhaps a JSON file could be made for those settings as well, though it should be fine to simply have them set to default when running from the command line.

The final command could look something like this:
```
 pixelstacker [--width=<width>] [--height=<height>] [--singlelayer] [--sideview] [--profile=<path>] <input> <output>
```

This is just a suggestion, of course. I don't know how much of this you'd be willing to implement (if anything at all, lol). In any case, thanks for you attention!

**Comment by Pangamma** [comment]
TBH I have been thinking about how cool it would be if people made videos using pixelstacker. I'll add this item into the backlog for sure. The spec on cmd arguments (and the expressed interest) will help a lot on this. Just be sure to share with me some of the stuff you make. I'd be interested in seeing what you can come up with.

**Comment by Pangamma** [comment]
Resolved in pr: #13 #13
